pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -299,47 +296,97 @@ func (s *Service) sendMsgEvents(msg *service.StateMsg) {
 func (s *Service) startInternalListener() {
 	go func() {
 		for msg := range s.callbackChannel {
-			// TODO handle error in callback - https://github.com/hyperledger/aries-framework-go/issues/242
-			if err := s.process(msg); err != nil {
-				// TODO handle error
-				logger.Errorf(err.Error())
+			if msg.Err != nil {
+				// TODO https://github.com/hyperledger/aries-framework-go/issues/438 Cleanup/Update data on Stop Event Action
+				logger.Errorf("client action event processing failed - msgID:%s error:%s", msg.ID, msg.Err)
+				err := s.processCallbackFailure(msg.ID, msg.Err)
+				if err != nil {
+					logger.Errorf("process callback failure: %s", err)
+				}
+				continue
+			}
+
+			// TODO https://github.com/hyperledger/aries-framework-go/issues/242 - retry logic
+
+			if err := s.process(msg.ID); err != nil {
+				procErr := s.processCallbackFailure(msg.ID, err)
```

Comment:
I expect this to be refactored in #242 such that the protocol instance isn't automatically abandoned due to some internal error

Reply:
Yes, here is the [TODO](https://github.com/hyperledger/aries-framework-go/pull/480/files#diff-f761c91e753eba1c42a68076df06ce12R306).